governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals.tsx

```diff
@@ -0,0 +1,501 @@
+  const { squads } = useMultisig(anchorWallet as Wallet)
+
+  useEffect(() => {
+    const fetchProposalInstructions = async () => {
+      const multisigParser = MultisigParser.fromCluster(
+        getMultisigCluster(cluster)
+      )
+      if (squads) {
+        const proposalIxs = []
+        for (let i = 1; i <= proposal.instructionIndex; i++) {
+          const instructionPda = getIxPDA(
+            proposal.publicKey,
+            new BN(i),
+            squads.multisigProgramId
+          )[0]
+          const instruction = await squads.getInstruction(instructionPda)
+          const parsedInstruction = multisigParser.parseInstruction({
+            programId: instruction.programId,
+            data: instruction.data as Buffer,
+            keys: instruction.keys as AccountMeta[],
+          })
+
+          proposalIxs.push(parsedInstruction)
+        }
+        setProposalInstructions(proposalIxs)
+      }
+    }
+
+    fetchProposalInstructions()
+  }, [proposal, squads, cluster])
+
+  return proposalInstructions ? (
+    <div className="grid grid-cols-3 gap-4">
+      <div className="col-span-3 my-2 space-y-4 bg-[#1E1B2F] p-4 lg:col-span-2">
+        <h4 className="h4">Info</h4>
+        <hr className="border-gray-700" />
+        <div className="flex justify-between">
+          <div>Proposal Pubkey</div>
+          <div>{proposal.publicKey.toBase58()}</div>
+        </div>
+        <div className="flex justify-between">
+          <div>Creator</div>
+          <div>{proposal.creator.toBase58()}</div>
+        </div>
+        <div className="flex justify-between">
+          <div>Multisig</div>
+          <div>{proposal.ms.toBase58()}</div>
+        </div>
+      </div>
+      <div className="col-span-3 my-2 space-y-4 bg-[#1E1B2F] p-4 lg:col-span-1">
+        <h4 className="h4 mb-4">Results</h4>
+        <hr className="border-gray-700" />
+        <div className="grid grid-cols-3 justify-center gap-4 pt-5 text-center align-middle">
+          <div>
+            <div className="font-bold">Confirmed</div>
+            <div className="text-lg">{proposal.approved.length}</div>
+          </div>
+          <div>
+            <div className="font-bold">Cancelled</div>
+            <div className="text-lg">{proposal.cancelled.length}</div>
+          </div>
+          <div>
+            <div className="font-bold">Threshold</div>
+            <div className="text-lg">
+              {multisig.threshold}/{multisig.keys.length}
+            </div>
+          </div>
+        </div>
+      </div>
+      <div className="col-span-3 my-2 space-y-4 bg-[#1E1B2F] p-4">
+        <h4 className="h4">Instructions</h4>
+        <hr className="border-gray-700" />
+        {proposalInstructions?.map((instruction, index) => (
+          <>
+            <div key={`${index}_instruction`} className="flex justify-between">
+              <div>Instruction</div>
+              <div>
+                {instruction instanceof PythMultisigInstruction ||
```

Comment:
Can you display that the name of the program.
`Program : Pyth oracle`
or 
`Program : Wormhole`